actions/aws-auth/README.md

```diff
@@ -21,21 +21,84 @@ jobs:
-| `pass-claims`              | String | `, `-separated list of [GitHub Actions claims](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect#understanding-the-oidc-token) (session tags) to make available to `role-arn`. Currently supported claims (default): `"repository_owner, repository_name, job_workflow_ref"` [^1] |
+| `role-arn`                 | String | Specify custom workload role. Role ARN must be prefixed with `github-actions` e.g. `arn:aws:iam::366620023056:role/github-actions/s3-test-access` [^1]                                |
+| `pass-claims`              | String | `, `-separated list of [GitHub Actions claims](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect#understanding-the-oidc-token) (session tags) to make available to `role-arn`. Currently supported claims (default): `"repository_owner, repository_name, job_workflow_ref"` [^2] |
 | `set-creds-in-environment` | Bool   | Set environment variables for AWS CLI and SDKs (default: `true`)                                                                                                                      |
 | `role-duration-seconds`    | String | Role duration in seconds (default: `"3600"`)                                                                                                                                          |
 
-[^1]: GitHub OIDC token claims must be mapped to the Cognito identity pool before they can be used. If you would like to use a claim that is not listed, file an issue in this repo or reach out to `@platform-productivity` in `#platform`.
+[^1]: See [Setting up Workload Role](#setting-up-workload-role) for an example
+[^2]: GitHub OIDC token claims must be mapped to the Cognito Identity Pool before they can be used. If you would like to use a claim that is not listed, file an issue in this repo or reach out to `@platform-productivity` in `#platform`.
 
 This uses the [`cognito-idpool-auth`](https://github.com/catnekaise/cognito-idpool-auth) action to perform authentication with an Amazon Cognito Identity Pool using the GitHub Actions OIDC access token.
+
+## Setting up Workload Role
+
+IAM workload roles are used to grant permissions to AWS in a secure manner. From a workflow run, once authenticated, the role is granted temporary credentials to access AWS resources permitted by the associated IAM role and attached trust/permission policies. The following steps will guide you through the process of setting up an IAM workload role for read access to a single object in an S3 bucket.
+
+### Create IAM Role
+
+Make sure when creating the role, the path is prefixed with `github-actions`. The Cognito Identity Pool only allows authenticated roles that match the following naming scheme: `"arn:aws:iam::*:role/github-actions/*"`
```

Comment:
Probably also worth mentioning roles have to be created in a different account and not in `grafanalabs-github-actions`?

Reply:
Yeah. We could mention that the role should be in the account where the resources it wants to access are.

Since there shouldn't be resources in `grafanalabs-github-actions`, that should take care of that part.